src/ax_interface/socket_io.py

```diff
@@ -25,6 +26,20 @@ def __init__(self, mib_table, run_event, loop):
 
         self.transport = self.ax_socket = None
 
+        self.ax_socket_path = constants.AGENTX_SOCKET_PATH
+        # open the snmpd config file and search for a agentx socke redefinition
+        pattern = re.compile("^agentxsocket\s+(.*)$")
```

Comment:
>pattern [](start = 8, length = 7)

There is a corner case. 'snmpd -x' will override this config. How do you handle it?

Reply:
true, I don't think there is a way to poll the daemon for autodiscovery of the AgentX address, we could use _psutils_ to look for the process and parse its 'cmdline' attribute. I think this is a corner case that should be documented that if you want to override the agentx socket use the config file.

Reply:
Do you need me to implement the commandline parsing or can we live without it?
I will add a code comment that points to the corner case for future reference

Reply:
No need if you add the code comment.

---
In reply to: [235185409](https://github.com/Azure/sonic-snmpagent/pull/92#discussion_r235185409) [](ancestors = 235185409)